back_ends/bash/exec.sh

```diff
@@ -1,12 +1,18 @@
 code_file_path="$1"
 test_code_file_path="$2"
+is_checking_code="$3"
 # We won't use this because this back end only supports text-based back ends.
 #output_type="$3"
 
-if [ -f "$test_code_file_path" ]
+if [[ "$is_checking_code" == "True" ]]
```

Comment:
This looks good, but please make sure you've tested it carefully in the app itself (if you haven't already).

Reply:
Alright, I tested the R and bash back_ends and they check the code correctly! Here are the checking_codes I used, if applicable. I guess I don't know what is meant by python script so I haven't checked those but the exec.py and exec.sh files are identical to the others. 
```
connection <- file("code")
open(connection)
line <- readLines(connection, n=1)
search <- "len"
while(length(line) > 0) {
  
  if(grepl(search, line)) {
    print("Error: len() function not allowed in this exercise")
  } 
  line <- readLines(connection, n=1)
}
close(connection)
```

```
FIND='#'
while read -r line;
do
if grep -q "$FIND" <<< "$line"; then
echo "Error: $FIND operator not allowed in this exercise"
fi
done < code
```

Reply:
A note on the bash backend, it could be that I chose a poor method for reading through the file but bash never read through the final line of a student's code if it didn't end in a newline, meaning everything but the final line was checked. I solved this by adding the line 
```
  sed -i -e '$a\' "$code_file_path"
```
to exec.sh, which adds a new_line to a student's code if there isn't one already. This fixes the problem and ensures the entire file is checked.

Reply:
That's good. If you haven't already, please leave a comment that indicates why you added this line.